integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonInsertFromStageCommand.scala

```diff
@@ -477,6 +487,48 @@ case class CarbonInsertFromStageCommand(
     output.asScala
   }
 
+  /**
+   * create '.loading' file to tag the stage in process
+   * Return false means the stage files were creat successfully
+   * While return true means the stage files were failed to create
+   */
+  private def createStageLoadingFiles(
+      executorService: ExecutorService,
+      stageFiles: Array[(CarbonFile, CarbonFile)]): Array[(CarbonFile, CarbonFile)] = {
+    stageFiles.map { files =>
+      executorService.submit(new Callable[Boolean] {
+        override def call(): Boolean = {
+          val stageLoadingFile =
+            FileFactory.getCarbonFile(files._1.getAbsolutePath +
+              CarbonTablePath.LOADING_FILE_SUBFIX);
+          if (!stageLoadingFile.exists()) {
```

Comment:
exists and createNewFile not in a same transaction, so createNewFile may be fail. suggest change to:
if (!stageLoadingFile.createNewFile()) {
  stageLoadingFile.setLastModifiedTime(System.currentTimeMillis());
}

Reply:
modified